packages/tenant-process/src/services/tenantService.ts

```diff
@@ -281,6 +283,37 @@ export function tenantServiceBuilder(
       });
     },
 
+    async deleteTenantMailById(
+      tenantId: TenantId,
+      mailId: string,
+      organizationId: TenantId,
+      correlationId: string
+    ): Promise<void> {
+      logger.info(`Deleting mail ${mailId} to Tenant ${tenantId}`);
+
+      const tenant = await retrieveTenant(tenantId, readModelService);
+
+      if (tenantId !== organizationId) {
+        throw operationForbidden;
+      }
+
+      const updatedTenant: Tenant = {
+        ...tenant.data,
+        mails: [...tenant.data.mails.filter((mail) => mail.id !== mailId)],
```

Comment:
We also need to check if the `mailId` exists within the tenant, otherwise throws `MailNotFound`